### PR TITLE
Fixes some warnings and a related bug in CCSprite.

### PR DESCRIPTION
**Pull request by xpol**
This story tells us that we should not just ignore warnings.

### CODE REVIEW

**Comment by pandamicro** [comment]
Thanks @xpol 
@VirtualSJ Please review the modifications in readers. Do we need to change the JSON parser too?

File: cocos/2d/CCSprite.cpp [hunk 1]
```diff
@@ -223,7 +223,7 @@ bool Sprite::initWithPolygon(const cocos2d::PolygonInfo &info)
 {
     Texture2D *texture = Director::getInstance()->getTextureCache()->addImage(info.filename);
     bool res = false;
-    if(initWithTexture(texture));
```
Comment (zilongshanren) [inline cocos/2d/CCSprite.cpp:226]: good catch!

**Comment by zilongshanren** [comment]
@xpol  Thanks :+1:

**Comment by zilongshanren** [comment]
@xpol  I also hate warnings, but now there are tons of warnings and many of them are in 3rd party libs..

**Comment by xpol** [comment]
@zilongshanren thanks. Lets get the warnings fixed.

**Pull request merged**